src/transformers/trainer.py

```diff
@@ -1158,6 +1158,26 @@ def call_model_init(self, trial=None):
 
         return model
 
+    def torch_jit_model_eval(self, model, training=False, dataloader=None):
+        if not training:
+            jit_inputs = []
+            example_batch = next(iter(dataloader))
```

Comment:
is `dataloader` required? As I commented before you're not checking anywhere if it's None. 

If it's required then please remove `=None` in the function signature.

Reply:
yes, it is required for jit.
Thanks for your suggestion, here have made changes to add a checking before constructing the jit inputs and removed `=None` in the function signature.

Reply:
All looks good now to me.

Thank you!